### PR TITLE
Adding Model Factories for Unit Test Mocking

### PR DESCRIPTION
**Pull request by minnieliu**
Adding model factories for all Communication SDKs.

Note that some model classes are internal and I wasn't able to include them in the public model factory because it would cause Inconsistent accessibility issues. 
For example:
image

### CODE REVIEW

File: sdk/communication/Azure.Communication.Administration/src/Models/AdministrationModelFactory.cs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+            new AcquiredPhoneNumbers(phoneNumbers, nextLink);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AreaCodes"/> class.
+        /// </summary>
+        /// <param name="primaryAreaCodes"> Represents the list of primary area codes. </param>
+        /// <param name="secondaryAreaCodes"> Represents the list of secondary area codes. </param>
+        /// <param name="nextLink"> Represents the URL link to the next page. </param>
+        /// <returns>A new <see cref="AreaCodes"/> instance for mocking.</returns>
+        public static AreaCodes AreaCodes(IReadOnlyList<string> primaryAreaCodes, IReadOnlyList<string> secondaryAreaCodes, string nextLink) =>
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/src/Models/AdministrationModelFactory.cs:40]: nit: let's move all of the `=>` to the new line after the method signature

File: sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs [hunk 2]
```diff
@@ -134,6 +134,35 @@ public partial class AcquiredPhoneNumbers
...
+        public static Azure.Communication.Administration.Models.PhonePlan PhonePlan(string phonePlanId, string localizedName, Azure.Communication.Administration.Models.LocationType locationType) { throw null; }
+        public static Azure.Communication.Administration.Models.PhonePlanGroup PhonePlanGroup(string phonePlanGroupId, string localizedName, string localizedDescription) { throw null; }
+        public static Azure.Communication.Administration.Models.PhonePlanGroups PhonePlanGroups(System.Collections.Generic.IReadOnlyList<Azure.Communication.Administration.Models.PhonePlanGroup> phonePlanGroupsValue, string nextLink) { throw null; }
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs:159]: Some of these types should not be public like `PhonePlanGroups`. Do you mind filing a work item to go through the models and mark the ones that are not needed as `internal`?

**Review by RezaJooyandeh — APPROVED** [review]
One nit comment, otherwise looks good to me. Thanks

File: sdk/communication/Azure.Communication.Administration/src/Models/AdministrationModelFactory.cs (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/Azure.Communication.Administration/src/Models/AdministrationModelFactory.cs]: Done, thanks!

**Comment by juancamilor** [comment]
@minnieliu @RezaJooyandeh @KrzysztofCwalina A bit late on this review, but what is the intend of all these factories, is it just because some models/ classes are not available to the test project due to their access modifiers ? If that is the only reason, we could easily do what other projects have done, add an exclusion on the assembly so the tests have access to it.
Sorry if this was discussed somewhere else but want to know the rational. Looks like an unnecessary complexity layer that will spread to all sdks in all languages likely.

**Comment by RezaJooyandeh** [comment]
@juancamilor the reason for this is to enable developer to mock our service when doing their own development otherwise they have build unnecessary wrappers. For a public function of ours to be mocked, the developers should be able to instantiate the return type. Model factories are the standard way that Azure libraries provide this functionality without directly adding a public constructor which might lead into unwanted usage patterns. See this for more details: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking

**Comment by juancamilor** [comment]
Thanks for the clarification and the reference in the docs @RezaJooyandeh . It makes sense, I thought the actual internal code was going to be refactored to used these factories, but it is clear to me now that it is just to provide support in testing code to be able to instantiate the models.

File: sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs (hunk 2 above)
Comment (minnieliu) [reply sdk/communication/Azure.Communication.Administration/api/Azure.Communication.Administration.netstandard2.0.cs:159]: Added the work item here: https://skype.visualstudio.com/SPOOL/_workitems/edit/2299174

**Pull request merged**